HaxeManual/10-std.tex

```diff
@@ -749,21 +749,96 @@ \subsection{Encoding Xml}
 trace(xml.toString()); // <root><child1/><child2/></root>
 \end{lstlisting}
 
+\subsection{Simplified Xml access}
+\label{std-Xml-simplified-access}
+
+The \href{https://api.haxe.org/haxe/xml/Access.html}{haxe.xml.Access} API exists to provide a dot-syntax access to the most common \type{Xml} methods.
+
+Since Haxe 4 this class is named \type{haxe.xml.Access}. In Haxe 3 it was known as \type{haxe.xml.Fast}.
+
+Here's an example of XML Access usage:
+\begin{lstlisting}
+// parse some XML data
+var xml = Xml.parse("
+  <user name='John'>
+    <phone>
+      <number>0000</number>
+      <number>111</number>
+    </phone>
+  </user>
+");
+// wrap the Xml for Access
+var access = new haxe.xml.Access(xml.firstElement());
+
+// access attributes
+trace(access.att.name); // attribute "name"
+if (access.has.age) trace( access.att.age ); // optional attribute
+
+// access the "phone" child, which is wrapped with haxe.xml.Access too
+var phone = access.node.phone;
+
+// iterate over numbers
+for (p in phone.nodes.number) {
+  trace(p.innerData);
+}
+\end{lstlisting}
+This code works the same on all platforms.
+
+\paragraph{Accessors}
+There are different accessors that can be used with the Access API:
+\begin{description}
+	\item[\expr{.name}] returns the name of the current element (same as \expr{Xml.nodeName}).
+	\item[\expr{.x}] returns the current corresponding \type{Xml} node.
+	\item[\expr{.att.<name>}] access to a given attribute. An exception is thrown if the attribute doesn't exists.
+	\item[\expr{.has.<name>}] check the existence of an attribute.
+	\item[\expr{.elements}] the list of all sub-elements (which are the nodes with type \type{Xml.Element}).
+	\item[\expr{.node.<name>}] access to the first sub element with the given name. An exception is thrown if the element doesn't exists.
+	\item[\expr{.nodes.<name>}] returns a List of elements with the given name.
+	\item[\expr{.hasNode.<name>}] check the existence of a sub node with the given name.
+	\item[\expr{.innerData}] returns the inner \ic{PCDATA} or \ic{CDATA} of the node. An exception is thrown if there is no data or if there not only data but also other nodes.
+	\item[\expr{.innerHTML}] returns the XML string built with all the sub nodes, excluding the current one.
+\end{description}
+
+\paragraph{Self-closed nodes}
+Please note that we cannot access self-closed nodes as "regular" ones. 
+For example, provided this XML:
+
+\begin{lstlisting}
+var xml = Xml.parse("
+  <myNode1></myNode1>
+  <myNode2/>
+");
+
+var access = new haxe.xml.Access(xml.firstElement());
```

Comment:
This example is borked because `xml.FirstElement()` would refer to the `myNode1` here. Might want to wrap the content that's parsed above in `<xml> ... </xml>` for this to make sense.

Reply:
It does work here: https://try.haxe.org/#330f2

Reply:
There it refers to the `<user>` node. Here it refers to the `<myNode1>` node and then you try to access `node.myNode1` which doesn't make sense.